src/test/miner_tests.cpp

```diff
@@ -242,7 +247,14 @@ void MinerTestingSetup::TestPackageSelection(
     tx.vout[0].nValue = (100000000 - 10000) * SATOSHI;
     m_node.mempool->addUnchecked(entry.Fee(10000 * SATOSHI).FromTx(tx));
     pblocktemplate = AssemblerForTest(config).CreateNewBlock(scriptPubKey);
-    BOOST_CHECK_EQUAL(pblocktemplate->block.vtx[3]->GetId(), lowFeeTxId2);
+    const auto foundTx = std::find_if(
+        pblocktemplate->block.vtx.begin(), pblocktemplate->block.vtx.end(),
+        [lowFeeTxId2](const CTransactionRef &a) -> bool {
+            return a->GetId() == lowFeeTxId2;
+        });
```

Comment:
Good. Thanks for cleaning this up finally

Reply:
This was surprisingly painful to figure out. Lack of defined operators on some of the Bitcoind types made my life difficult.